# awesome-transformer-bio

A curated list of awesome transformer applications in the field of computational biology.

## 2025

- **2025** | DrBioRight 2.0: an LLM-powered bioinformatics chatbot for large-scale cancer functional proteomics analysis | [Nature Communications](https://www.nature.com/articles/s41467-025-57430-4#citeas)
- **2025** | Learning and actioning general principles of cancer cell drug sensitivity | [Nature Communications](https://www.nature.com/articles/s41467-025-56827-5)
- **2025** | Evaluation of large language models for discovery of gene set function | [Nature Methods](https://www.nature.com/articles/s41592-024-02525-x)
- **2025** | A joint analysis of single cell transcriptomics and proteomics using transformer | [npj Systems Biology and Applications](https://www.nature.com/articles/s41540-024-00484-9)
- **2025** | A cell atlas foundation model for scalable search of similar human cells | [Nature](https://www.nature.com/articles/s41586-024-08411-y)
- **2025** | Pretrained transformers applied to clinical studies improve predictions of treatment efficacy and associated biomarkers | [Nature Communications](https://www.nature.com/articles/s41467-025-57181-2)
- **2025** | Transformers and genome language models | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-025-01007-9)
- **2025** | Interpretable identification of cancer genes across biological networks via transformer-powered graph representation learning | [Nature Biomedical Engineering](https://www.nature.com/articles/s41551-024-01312-5)
- **2025** | Nucleotide Transformer: building and evaluating robust foundation models for human genomics | [Nature Methods](https://www.nature.com/articles/s41592-024-02523-z)
- **2025** | FeatureDock for protein-ligand docking guided by physicochemical feature-based local environment learning using transformer | [npj Drug Discovery](https://www.nature.com/articles/s44386-025-00005-6)
- **2025** | An interpretable RNA foundation model for exploring functional RNA motifs in plants | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00946-z)
- **2025** | Transformer-based modeling of Clonal Selection and Expression Dynamics reveals resistance mechanisms in breast cancer | [npj Systems Biology and Applications](https://www.nature.com/articles/s41540-024-00485-8#Sec9)
- **2025** | NuFold: end-to-end approach for RNA tertiary structure prediction with flexible nucleobase center representation | [Nature Communications](https://www.nature.com/articles/s41467-025-56261-7)
- **2025** | Multimodal histopathologic models stratify hormone receptor-positive early breast cancer | [Nature Communications](https://www.nature.com/articles/s41467-025-57283-x)
- **2025** | Characterization of tumour heterogeneity through segmentation-free representation learning on multiplexed imaging data | [Nature Biomedical Engineering](https://www.nature.com/articles/s41551-025-01348-1)
- **2025** | Predicting RNA-seq coverage from DNA sequence as a unifying model of gene regulation | [Nature Genetics](https://www.nature.com/articles/s41588-024-02053-6)
- **2025** | RNA-protein interaction prediction using network-guided deep learning | [Communications Biology](https://www.nature.com/articles/s42003-025-07694-9#Sec16)
- **2025** | CelloType: a unified model for segmentation and classification of tissue images | [Nature Methods](https://www.nature.com/articles/s41592-024-02513-1)
- **2025** | A DNA language model based on multispecies alignment predicts the effects of genome-wide variants | [Nature Biotechnology](https://www.nature.com/articles/s41587-024-02511-w)
- **2025** | A foundation model of transcription across human cell types | [Nature](https://www.nature.com/articles/s41586-024-08391-z)
- **2025** | Contrastive-learning of language embedding and biological features for cross modality encoding and effector prediction | [Nature Communications](https://www.nature.com/articles/s41467-025-56526-1)
- **2025** | From computational models of the splicing code to regulatory mechanisms and therapeutic implications | [Nature Reviews Genetics](https://www.nature.com/articles/s41576-024-00774-2)
- **2025** | ARNLE model identifies prevalence potential of SARS-CoV-2 variants | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00919-2)
- **2025** | Paying attention to the SARS-CoV-2 dialect : a deep neural network approach to predicting novel protein mutations | [Communications Biology](https://www.nature.com/articles/s42003-024-07262-7#Sec16)
- **2025** | Attention-based deep learning for accurate cell image analysis | [Scientific Reports](https://www.nature.com/articles/s41598-025-85608-9)
- **2025** | A unified cross-attention model for predicting antigen binding specificity to both HLA and TCR molecules | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00973-w)
- **2025** | DNA promoter task-oriented dictionary mining and prediction model based on natural language technology | [Scientific Reports](https://www.nature.com/articles/s41598-024-84105-9)
- **2025** | Self-supervision advances morphological profiling by unlocking powerful image representations | [Scientific Reports](https://www.nature.com/articles/s41598-025-88825-4)
## 2024 

- **2024** | Discovering CRISPR-Cas system with self-processing pre-crRNA capability by foundation models | [Nature Communications](https://www.nature.com/articles/s41467-024-54365-0)
- **2024** | Large-language models facilitate discovery of the molecular signatures regulating sleep and activity | [Nature Communications](https://www.nature.com/articles/s41467-024-48005-w)
- **2024** | CancerGPT for few shot drug pair synergy prediction using large pretrained language models | [npj Digital Medicine](https://www.nature.com/articles/s41746-024-01024-9#citeas)
- **2024** | Simple and effective embedding model for single-cell biology built from ChatGPT | [Nature Biomedical Engineering](https://www.nature.com/articles/s41551-024-01284-6)
- **2024** | Optimization of hepatological clinical guidelines interpretation by large language models: a retrieval augmented generation-based framework | [npj Digital Medicine](https://www.nature.com/articles/s41746-024-01091-y)
- **2024** | Foundation models build on ChatGPT tech to learn the fundamental language of biology | [Nature Biotechnology](https://www.nature.com/articles/s41587-024-02400-2)
- **2024** | The model student: GPT-4 performance on graduate biomedical science exams | [Scientific Reports](https://www.nature.com/articles/s41598-024-55568-7)
- **2024** | Transformers in single-cell omics: a review and new perspectives | [Nature Methods](https://www.nature.com/articles/s41592-024-02353-z)
- **2024** | Improving prediction performance of general protein language model by domain-adaptive pretraining on DNA-binding protein | [Nature Communications](https://www.nature.com/articles/s41467-024-52293-7)
- **2024** | Multi role ChatGPT framework for transforming medical data analysis | [Scientific Reports](https://www.nature.com/articles/s41598-024-64585-5)
- **2024** | Single-nucleus and bulk RNA sequencing reveal cellular and transcriptional mechanisms underlying lipid dynamics in high marbled pork | [npj Science of Food](https://www.nature.com/articles/s41538-023-00203-4)
- **2024** | Scientific figures interpreted by ChatGPT: strengths in plot recognition and limits in color perception | [npj Precision Oncology](https://www.nature.com/articles/s41698-024-00576-z)
- **2024** | Large-scale foundation model on single-cell transcriptomics | [Nature Methods](https://www.nature.com/articles/s41592-024-02305-7)
- **2024** | DNA language model GROVER learns sequence context in the human genome | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00872-0)
- **2024** | A foundational large language model for edible plant genomes | [Communications Biology](https://www.nature.com/articles/s42003-024-06465-2)
- **2024** | scGraphformer: unveiling cellular heterogeneity and interactions in scRNA-seq data using a scalable graph transformer network | [Communications Biology](https://www.nature.com/articles/s42003-024-07154-w)
- **2024** | Protein embeddings predict binding residues in disordered regions | [Scientific Reports](https://www.nature.com/articles/s41598-024-64211-4)
- **2024** | Sliding-attention transformer neural architecture for predicting T cell receptor–antigen–human leucocyte antigen binding | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00901-y)
- **2024** | Deep learning prediction of ribosome profiling with Translatomer reveals translational regulation and interprets disease variants | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00915-6)
- **2024** | Rockfish: A transformer-based model for accurate 5-methylcytosine prediction from nanopore sequencing | [Nature Communications](https://www.nature.com/articles/s41467-024-49847-0)
- **2024** | RNA language models predict mutations that improve RNA function | [Nature Communications](https://www.nature.com/articles/s41467-024-54812-y)
- **2024** | Accurate RNA 3D structure prediction using a language model-based deep learning approach | [Nature Methods](https://www.nature.com/articles/s41592-024-02487-0)
- **2024** | MarsGT: Multi-omics analysis for rare population inference using single-cell graph transformer | [Nature Communications](https://www.nature.com/articles/s41467-023-44570-8)
- **2024** | Multi-purpose RNA language modelling with motif-aware pretraining and type-guided fine-tuning | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00836-4)
- **2024** | Digital profiling of gene expression from histology images with linearized attention | [Nature Communications](https://www.nature.com/articles/s41467-024-54182-5)
- **2024** | SpliceTransformer predicts tissue-specific splicing linked to human diseases | [Nature Communications](https://www.nature.com/articles/s41467-024-53088-6)
- **2024** | Transformers significantly improve splice site prediction | [Communications Biology](https://www.nature.com/articles/s42003-024-07298-9)
- **2024** | HBCVTr: an end-to-end transformer with a deep neural network hybrid model for anti-HBV and HCV activity predictor from SMILES | [Scientific Reports](https://www.nature.com/articles/s41598-024-59933-4)
- **2024** | Self-supervised learning for characterising histomorphological diversity and spatial RNA expression prediction across 23 human tissue types | [Nature Communications](https://www.nature.com/articles/s41467-024-50317-w)
- **2024** | scGPT: toward building a foundation model for single-cell multi-omics using generative AI | [Nature Methods](https://www.nature.com/articles/s41592-024-02201-0)
- **2024** | Precious2GPT: the combination of multiomics pretrained transformer and conditional diffusion for artificial multi-omics multi-species multi-tissue sample generation | [npj Aging](https://www.nature.com/articles/s41514-024-00163-3#citeas)
- **2024** | Multi-modal generative modeling for joint analysis of single-cell T cell receptor and gene expression data | [Nature Communications](https://www.nature.com/articles/s41467-024-49806-9)
- **2024** | GEFormerDTA: drug target affinity prediction based on transformer graph for early fusion | [Scientific Reports](https://www.nature.com/articles/s41598-024-57879-1)
- **2024** | scGAA: a general gated axial-attention model for accurate cell-type annotation of single-cell RNA-seq data | [Scientific Reports](https://www.nature.com/articles/s41598-024-73356-1)
- **2024** | Towards designing improved cancer immunotherapy targets with a peptide-MHC-I presentation model, HLApollo | [Nature Communications](https://www.nature.com/articles/s41467-024-54887-7)
- **2024** | StructmRNA a BERT based model with dual level and conditional masking for mRNA representation | [Scientific Reports](https://www.nature.com/articles/s41598-024-77172-5)
- **2024** | A 5′ UTR language model for decoding untranslated regions of mRNA and function predictions | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00823-9)
- **2024** | SCS: cell segmentation for high-resolution spatial transcriptomics | [Nature Methods](https://www.nature.com/articles/s41592-023-01939-3)
- **2024** | Deeper evaluation of a single-cell foundation model | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00949-w)
- **2024** | Automated real-world data integration improves cancer outcome prediction | [Nature](https://www.nature.com/articles/s41586-024-08167-5)
- **2024** | A community effort to optimize sequence-based deep learning models of gene regulation | [Nature Biotechnology](https://www.nature.com/articles/s41587-024-02414-w)
- **2024** | Generalized AI models for genomics applications | [Nature Methods](https://www.nature.com/articles/s41592-024-02524-y)
- **2024** | Zero-shot prediction of mutation effects with multimodal deep representation learning guides protein engineering | [Cell Research](https://www.nature.com/articles/s41422-024-00989-2)
- **2024** | Prediction of recurrence risk in endometrial cancer with multimodal deep learning | [Nature Medicine](https://www.nature.com/articles/s41591-024-02993-w)
- **2024** | SOFB is a comprehensive ensemble deep learning approach for elucidating and characterizing protein-nucleic-acid-binding residues | [Communications Biology](https://www.nature.com/articles/s42003-024-06332-0)
- **2024** | Language models for biological research: a primer | [Nature Methods](https://www.nature.com/articles/s41592-024-02354-y)
- **2024** | Detecting anomalous anatomic regions in spatial transcriptomics with STANDS | [Nature Communications](https://www.nature.com/articles/s41467-024-52445-9)
- **2024** | Predicting cell type-specific epigenomic profiles accounting for distal genetic effects | [Nature Communications](https://www.nature.com/articles/s41467-024-54441-5)
- **2024** | CRISPR-GPT: An LLM Agent for Automated Design of  Gene-Editing Experiments | [arxiv](https://arxiv.org/pdf/2404.18021v1)
- **2024** | DeepFM-Crispr: Prediction of CRISPR On-Target  Effects via Deep Learning | [NLP4Science](https://aclanthology.org/2024.nlp4science-1.2/)
- **2024** | Discovering CRISPR-Cas system with self-processing pre-crRNA capability by foundation models | [Nature Communications](https://www.nature.com/articles/s41467-024-54365-0)
- **2024** | RNA-GPT: Multimodal Generative System for RNA Sequence Understanding | [Workshop, NeurIPS](https://arxiv.org/abs/2411.08900)
- **2024** | LangCell: language-cell pre-training for cell identity understanding | [arXiv](https://arxiv.org/abs/2405.06708)
- **2024** | scMulan: a multitask generative pre-trained language model for single-cell analysis | [BioRxiv](https://www.biorxiv.org/content/10.1101/2024.01.25.577152v1)
- **2024** | GeneCompass: deciphering universal gene regulatory mechanisms with a knowledge-informed cross-species foundation model | [Cell Research](https://www.nature.com/articles/s41422-024-01034-y)
- **2024** | scmFormer Integrates Large-Scale Single-Cell Proteomics and Transcriptomics Data by Multi-Task Transformer | [Advanced Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/advs.202307835)
- **2024** | RiNALMo: General-Purpose RNA Language Models Can Generalize Well on Structure Prediction Tasks | [arXiv](https://arxiv.org/abs/2403.00043)
- **2024** | Delineating the effective use of self-supervised learning in single-cell genomics | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00934-3)
- **2024** | A long-context language model for deciphering and generating bacteriophage genomes | [Nature Communications](https://www.nature.com/articles/s41467-024-53759-4)
- **2024** | Accurate prediction of RNA secondary structure including pseudoknots through solving minimum-cost flow with learned potentials | [Communications Biology](https://www.nature.com/articles/s42003-024-05952-w)
- **2024** | Generative language models on nucleotide sequences of human genes | [Scientific Reports](https://www.nature.com/articles/s41598-024-72512-x)
- **2024** | Deep learning prediction of glycopeptide tandem mass spectra powers glycoproteomics | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00875-x)
- **2024** | Evaluating generalizability of artificial intelligence models for molecular datasets | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00931-6)
- **2024** | Codon language embeddings provide strong signals for use in protein engineering | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00791-0)
- **2024** | Riboformer: a deep learning framework for predicting context-dependent translation dynamics | [Nature Communications](https://www.nature.com/articles/s41467-024-46241-8)
- **2024** | Deciphering cell types by integrating scATAC-seq data with genome sequences | [Nature Computational Science](https://www.nature.com/articles/s43588-024-00622-7)
- **2024** | Reply to: Deeper evaluation of a single-cell foundation model | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00948-x)
- **2024** | Annotating protein functions via fusing multiple biological modalities | [Communications Biology](https://www.nature.com/articles/s42003-024-07411-y)
- **2024** | OpenFold: retraining AlphaFold2 yields new insights into its learning mechanisms and capacity for generalization | [Nature Methods](https://www.nature.com/articles/s41592-024-02272-z)
- **2024** | Large language models improve annotation of prokaryotic viral proteins | [Nature Microbiology](https://www.nature.com/articles/s41564-023-01584-8) 
- **2024** | State-specific protein–ligand complex structure prediction with a multiscale deep generative model | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-024-00792-z)

## 2023

- **2023** | Transformer for one stop interpretable cell type annotation | [Nature Communications](https://www.nature.com/articles/s41467-023-35923-4)
- **2023** | DeepConsensus improves the accuracy of sequences with a gap-aware sequence transformer | [Nature Biotechnology](https://www.nature.com/articles/s41587-022-01435-7)
- **2023** | Classification of deep-sea cold seep bacteria by transformer combined with Raman spectroscopy | [Scientific Reports](https://www.nature.com/articles/s41598-023-28730-w)
- **2023** | Reusability report: Learning the transcriptional grammar in single-cell RNA-sequencing data using transformers | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-023-00757-8)
- **2023** | Dynamic characterization and interpretation for protein-RNA interactions across diverse cellular conditions using HDRNet | [Nature Communications](https://www.nature.com/articles/s41467-023-42547-1#Sec13)
- **2023** | Multiclass malaria parasite recognition based on transformer models and a generative adversarial network | [Scientific Reports](https://www.nature.com/articles/s41598-023-44297-y)
- **2023** | Learning to predict RNA sequence expressions from whole slide images with applications for search and classification | [Communications Biology](https://www.nature.com/articles/s42003-023-04583-x)
- **2023** | ChatGPT for shaping the future of dentistry: the potential of multi-modal large language model | [International Journal of Oral Science](https://www.nature.com/articles/s41368-023-00239-y)
- **2023** | trRosettaRNA: automated prediction of RNA 3D structure with transformer network | [Nature Communications](https://www.nature.com/articles/s41467-023-42528-4)
- **2023** | SiGra: single-cell spatial elucidation through an image-augmented graph transformer | [Nature Communications](https://www.nature.com/articles/s41467-023-41437-w)
- **2023** | Single-cell biological network inference using a heterogeneous graph transformer | [Nature Communications](https://www.nature.com/articles/s41467-023-36559-0)
- **2023** | PeSTo: parameter-free geometric deep learning for accurate prediction of protein binding interfaces | [Nature Communications](https://www.nature.com/articles/s41467-023-37701-8)
- **2023** | Evaluating efficiency and accuracy of deep-learning-based approaches on study selection for psychiatry systematic reviews | [Nature Mental Health](https://www.nature.com/articles/s44220-023-00109-w)
- **2023** | Generative pretraining from large-scale transcriptomes for single-cell deciphering | [Iscience](https://www.cell.com/iscience/fulltext/S2589-0042(23)00613-2)
- **2023** | Transfer learning enables predictions in network biology | [Nature](https://www.nature.com/articles/s41586-023-06139-9)
- **2023** | CellPLM: pre-training of cell language model beyond single cells | [BioRxiv](https://www.biorxiv.org/content/10.1101/2023.10.03.560734v1)
- **2023** | CodonBERT: Large Language Models for mRNA design and optimization | [BioRxiv](https://www.biorxiv.org/content/10.1101/2023.09.09.556981v1)

## 2022

- **2022** | Transformer-based protein generation with regularized latent space optimization | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-022-00532-1)
- **2022** | scBERT as a large-scale pretrained deep language model for cell type annotation of single-cell RNA-seq data | [Nature Machine Intelligence](https://www.nature.com/articles/s42256-022-00534-z)
- **2022** | Learning the histone codes with large genomic windows and three-dimensional chromatin interactions using transformer | [Nature Communications](https://www.nature.com/articles/s41467-022-34152-5)
- **2022** | Interpretable RNA Foundation Model from Unannotated Data for Highly Accurate RNA Structure and Function Predictions | [ICML](https://arxiv.org/abs/2204.00300)
